pkg/scheduler/core/generic_scheduler.go

```diff
@@ -657,29 +657,37 @@ func PrioritizeNodes(
 		results[i] = make(schedulerapi.HostPriorityList, len(nodes))
 	}
 
-	processNode := func(index int) {
-		nodeInfo := nodeNameToInfo[nodes[index].Name]
+	// DEPRECATED: we can remove this when all priorityConfigs implement the
+	// Map-Reduce pattern.
+	workqueue.ParallelizeUntil(context.TODO(), 16, len(priorityConfigs), func(i int) {
+		priorityConfig := priorityConfigs[i]
+		if priorityConfig.Function == nil {
+			return
```

Comment:
One nit: we can remove L656~L658, and move L657 here:

```suggestion
			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
			return
```

Reply:
Done.

Reply:
NVM, this ParallelizeUntil runs for each priorityConfig, not for each node. Confused it with the next one.

Reply:
But for a particular `i` (the `i` here is for each `priorityConfig`, not `node`), it only runs once, isn't it?

Unlike the second `ParallelizeUntil` below, it's just iterating `priorityConfigs`.

Reply:
Yeap, we commented exactly at the same time :)